tests/filecheck/transforms/csl-stencil-to-csl-wrapper.mlir

```diff
@@ -0,0 +1,95 @@
+// RUN: xdsl-opt %s -p "csl-stencil-to-csl-wrapper" | filecheck %s
+
+builtin.module {
+  func.func @gauss_seidel(%a : !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>>, %b : !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>>) {
+    %0 = stencil.load %a : !stencil.field<[-1,1023]x[-1,511]xtensor<512xf32>> -> !stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>
+    %1 = "csl_stencil.prefetch"(%0) <{"topo" = #dmp.topo<1022x510>, "swaps" = [#csl_stencil.exchange<to [1, 0]>, #csl_stencil.exchange<to [-1, 0]>, #csl_stencil.exchange<to [0, 1]>, #csl_stencil.exchange<to [0, -1]>]}> : (!stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>) -> memref<4xtensor<510xf32>>
+    %2 = tensor.empty() : tensor<510xf32>
+    %3 = csl_stencil.apply(%0 : !stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>, %2 : tensor<510xf32>) <{"swaps" = [#csl_stencil.exchange<to [1, 0]>, #csl_stencil.exchange<to [-1, 0]>, #csl_stencil.exchange<to [0, 1]>, #csl_stencil.exchange<to [0, -1]>], "topo" = #dmp.topo<1022x510>, "num_chunks" = 2 : i64}> -> (!stencil.temp<[0,1]x[0,1]xtensor<510xf32>>) ({
+    ^0(%4 : memref<4xtensor<255xf32>>, %5 : index, %6 : tensor<510xf32>):
+      %7 = csl_stencil.access %4[1, 0] : memref<4xtensor<255xf32>>
+      %8 = csl_stencil.access %4[-1, 0] : memref<4xtensor<255xf32>>
+      %9 = csl_stencil.access %4[0, 1] : memref<4xtensor<255xf32>>
+      %10 = csl_stencil.access %4[0, -1] : memref<4xtensor<255xf32>>
+      %11 = arith.addf %8, %7 : tensor<255xf32>
+      %12 = arith.addf %10, %9 : tensor<255xf32>
+      %13 = arith.addf %12, %11 : tensor<255xf32>
+      %14 = "tensor.insert_slice"(%13, %6, %5) <{"static_offsets" = array<i64: 0>, "static_sizes" = array<i64: 255>, "static_strides" = array<i64: 1>, "operandSegmentSizes" = array<i32: 1, 1, 1, 0, 0>}> : (tensor<255xf32>, tensor<510xf32>, index) -> tensor<510xf32>
+      csl_stencil.yield %14 : tensor<510xf32>
+    }, {
+    ^1(%15 : !stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>, %16 : tensor<510xf32>):
+      %17 = csl_stencil.access %15[0, 0] : !stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>
+      %18 = csl_stencil.access %15[0, 0] : !stencil.temp<[-1,2]x[-1,2]xtensor<512xf32>>
```

Comment:
Is there any particular reason why we need two access ops here?

Reply:
We could canonicalise it at an earlier stage, it's a relic from the tensorisation pass.